cachi2/core/package_managers/bundler/main.py

```diff
@@ -26,3 +31,30 @@ def _resolve_bundler_package(package_dir: RootedPath, output_dir: RootedPath) ->
     """Process a request for a single bundler package."""
     parse_lockfile(package_dir)
     return []
+
+
+def _prepare_for_hermetic_build(output_dir: RootedPath) -> None:
+    """Prepare a package for hermetic build by injecting necessary config."""
+    bundle_config = output_dir.join_within_root(".bundle/config").path
+    hermetic_config = dedent(
+        f"""
+        BUNDLE_CACHE_PATH={output_dir}/deps/rubygems
+        BUNDLE_DEPLOYMENT=true
+        BUNDLE_NO_PRUNE=true
+    """
+    )
+    if bundle_config.exists():
```

Comment:
Also, the config file is not mandatory.
https://github.com/3scale/porta  -  There is no `.bundle/config`

Reply:
Yes, but it could be present and if it is present fir any reason then we do not want to just overwrite it.

Reply:
> we do not want to just overwrite it.

I believe we have no choice, we had a long discussion on that in the design discussion, and it's due to the order of precedence where config takes priority over variables. If there are multiple ruby packages in a repo we can't use `$BUNDLE_APP_CONFIG` because it would be global for all packages we process, unfortunately, but each package can define their own options, so we have to override them individually, it's weird, it's wrong, but I don't think we've come up with a different idea in the design, have we?

Reply:
Actually, I have an alternative proposal. Since it's inherently wrong to dirty users' repos we could document this as a known limitation of our project. IOW, we would say that while we support compound repos, in case users use a repo with more than a single Ruby package it might happen (depending on whether `.bundle/config`s were checked in the repo for the respective packages) that during the hermetic build we might use wrong bundler configuration. <<< This is based on the fact that encountering mono-language compound repos is a fairly rare case so realistically a miniscule percentage of the user base would be actually affected. We could then make use of the `$BUNDLE_APP_CONFIG` variable. That should IMO be fine for some initial implementation of bundler support.

Then, as a follow up, we should think of enhancing our core architecture by either:
- having a per-package build_config.json OR
- having a single build_config.json, but namespace the env vars and settings we need to apply **per-package**, so in practical terms in Bundler's case we'd have a set of config options for package A and a different set for package B. We'd then have to document that users have to do `source cachi2-output/<pkg>/cachi2.env` N times depending on how many packages (N) they're trying to build in their environment. That way we could define the `$BUNDLE_APP_CONFIG` multiple times without having to worry that it would override some other package's bundler config.
Like I said ^this is a follow up material and we should track it as an issue, because this is not the first time we're seeing this problem, the same is happening with extra input JSON options for the rpm backend.

Reply:
So in the end it looks like having the variables set would effectively cover the most frequent use-cases, at least for now. Everything else is probably unlikely enough. I'll do that in the next revision.

Reply:
Searching github for the variables that we set resulted in less than a hundred matches for all three of them out of more than a million repositories. I think this confirms that while possible this failure scenario is not very probable.

Reply:
> I think this confirms that while possible this failure scenario is not very probable.

It's not about the variables we set, it's more about using **some** set of variables coming from package A and using it to build package B, so it's very particular mix of events that would lead to the problem. That said, we currently don't have any such limitation on compound repos with other pkg mgrs, so we need to be open about this, yet small, likelihood of incorrect variable usage.
@a-ovchinnikov in general, did the package namespaced settings idea in the build config make sense to you?

Reply:
Yes, namespaces make sense as a general solution to the problem. I am not sure we have this problem now or will have it in the near future because I switched to setting environment variables and those do not seem to clash much with anything found in a local config (judging from a Github search).